services/actions/schedule.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+
+	actions_model "code.gitea.io/gitea/models/actions"
+	"code.gitea.io/gitea/modules/log"
+
+	"github.com/robfig/cron/v3"
+)
+
+var schedule *Schedule
+
+type Schedule struct {
+	c *cron.Cron
+}
+
+func (s *Schedule) AddFunc(spec string, f func()) (int, error) {
+	id, err := s.c.AddFunc(spec, f)
+	return int(id), err
+}
+
+func (s *Schedule) Remove(ids []int) {
+	for _, id := range ids {
+		s.c.Remove(cron.EntryID(id))
+	}
+}
+
+func newSchedule() {
+	c := cron.New()
+	c.Start()
+	schedule = &Schedule{
+		c: c,
+	}
+}
+
+func resetSchedule() {
```

Comment:
After reading the [code](https://github.com/robfig/cron/blob/bc59245fe10efaed9d51b56900192527ed733435/cron.go#L158), I understand why we should do this, there's no way to specify the id of the new entry.

But I still think it's not a good idea to reset all schedules in DB.

It doesn't block, maybe we can improve it in the future.